server/gui/main.py

```diff
@@ -155,14 +154,15 @@ def __init__(self, parent):
         self.signals.selectedFile.connect(self.createScanpyEngine)
 
     def updateEmbedding(self, idx):
-        self.embedding_selection = MODES[idx]
+        self.embedding_selection = [MODES[idx]]
 
     def createScanpyEngine(self, file_name):
-        worker = DataLoadWorker(file_name, self.embedding_selection)
-        worker.signals.result.connect(self.onDataSuccess)
-        worker.signals.error.connect(self.onDataError)
-        self.load_worker = threading.Thread(target=worker.run, daemon=True)
-        self.load_worker.start()
+        worker = Worker(self.window().child_conn, file_name, self.title, host="127.0.0.1", port=GUI_PORT,
```

Comment:
See comment in Emitter about keeping both ends of the pipe open in both processes.